lib/core/test/unit/Cardano/Pool/PerformanceSpec.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Cardano.Pool.PerformanceSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Pool.Performance
+    ( type (>=) (Bounded) )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.QuickCheck
+import Test.QuickCheck.Arbitrary.Generic
+    ( genericArbitrary, genericShrink )
+
+import qualified Cardano.Pool.Performance as P
+
+spec :: Spec
+spec = do
+    describe "nonMyopicMemberRewards" $ do
+        describe "all else equal..." $ do
+            it "lower margin => higher desirability (or equal)" $
+                property $
+                    allElseEqualProperty
+                        "margin" (P.margin) (\x y -> x{P.margin = y})
+                        (show . P.unMargin)
+                        (<)
+
+            it "lower cost => higher desirability (or equal)" $
+                property $ do
+                    allElseEqualProperty
+                        "cost" (P.cost) (\x y -> x{P.cost = y})
+                        (show . P.getLovelace)
+                        (<)
+
+            it "higher performance => higher desirability (or equal)" $
+                property $
+                    allElseEqualProperty
+                        "performance"
+                        (P.recentAvgPerformance)
+                        (\x y -> x{P.recentAvgPerformance = y})
+                        (show . P.unBounded)
+                        (>)
+
+-- | If two pools differ only in one field, and one is more desirable than the
+-- other, how should the field values compare?
+allElseEqualProperty
```

Comment:
Sadly the important `(<)` `(>)` distinction on the call-sites ended up less readable than I'd hoped.

Reply:
Think I can simplify the structure a bit. Will do tomorrow.